buuf-icon-theme: init at 3.45

sk4rd opened the pull request:
## Description of changes

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Commit:
maintainers: add sk4rd

Commit:
buuf-icon-theme: init at 3.45

h7x4 commented on pkgs/data/icons/buuf-icon-theme/default.nix:
```diff
@@ -0,0 +1,24 @@
+  pname = "buuf-icon-theme";
+  version = "3.45";
+
+  src = fetchurl {
```
I believe the correct fetcher in this case would be `fetchzip`

```suggestion
  src = fetchzip {
```

h7x4 commented on pkgs/data/icons/buuf-icon-theme/default.nix:
```diff
@@ -0,0 +1,24 @@
+    sha256 = "sha256-w6ertMv3ERkj9brJsxodo5JbaPeS0a29SOVkUK+o7MU=";
+  };
+
+  installPhase = ''
+    mkdir -p $out/share/icons/
+    cp -r . $out/share/icons/${pname}
+  '';
```
Don't forget the `runHooks`s (see checklist): https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#new-packages

I also don't think the location of the icons depends on the name of the package, just because they happen to be the same

```suggestion
  installPhase = ''
    runHook preInstall
    
    mkdir -p $out/share/icons/
    cp -r . $out/share/icons/buuf-icon-theme
    
    runHook postInstall
  '';
```

h7x4 commented on pkgs/data/icons/buuf-icon-theme/default.nix:
```diff
@@ -0,0 +1,24 @@
+  '';
+
+  meta = with lib; {
+    description = "Buuf icon theme";
```
This description doesn't say much more than the package name. How about something like this?

```suggestion
    description = "A comprehensive icon theme adapted for Gnome";
```

h7x4 reviewed (commented):
Hello @sk4rd, welcome to nixpkgs!

I've left a few comments below. Also, the commit where you add yourself to the maintainer list needs to be renamed to say `maintainers: add sk4rd`. Else everything looks good!

sk4rd commented:
Hey @h7x4,
I have applied the tips from your comments to the derivation.
I was unsure about picking the correct message for the new commit.
Is it okay?

h7x4 commented:
Right! Commits in nixpkgs are generally very coarse. We squash all commits into logical units that does one single thing, no matter how big or small. So it'd be great if you could squash the fixup commits into the `buuf-icon-theme: init at 3.45` commit, since these are technically part of creating a new package.

sk4rd commented:
I've reworked the commit structure so that it corresponds to the standard in nixpkgs.
Additionally, I've deleted the unnecessary `rec` keyword.
I'm sorry for my previous misunderstandings, and thank you for helping me out.

sk4rd commented:
Wow! Still got a lot of learning to do...
I have squashed all the commits into `buuf-icon-theme: init at 3.45` and also deleted the redundant `rec` keyword.
I'm sorry for my previous misunderstandings, and thank you for helping me out!

h7x4 commented:
No worries :grin:

Since this is your first contribution, I'll assume you're not that familiar with the process of getting things merged here. It can sometimes require quite a bit of patience, as there are a lot more PRs than there are available reviewers and especially committers (~200 people with varying degrees of spare time).

There are some ways to speed up the process. There are some discourse threads named `PRs ready for {reviews,testing,*}`, where you can post a link and hope someone will review it. There's also the matrix dev channels, where you can chat with people and ask questions.

Alternatively, if you feel like you're up for a challenge, we'd really appreciate help with reviewing other PRs as well! There are several ways to help review PRs without deep knowledge of nix, NixOS or nixpkgs, for example by building software in other PRs using [nixpkgs-review](https://github.com/Mic92/nixpkgs-review), and making sure it works. This would help speed up the review process for everyone, but it's completely optional of course.

h7x4 commented on pkgs/data/icons/buuf-icon-theme/default.nix:
```diff
@@ -0,0 +1,28 @@
+
+  src = fetchzip {
+    url = "http://buuficontheme.free.fr/buuf3.45.tar.xz";
+    sha256 = "sha256-TrUHW3W4fMR8esirEenKxV+cxnzjC5U0raPfutear+g=";
```
`hash` and SRI hashes are the way these are used these days.

```suggestion
    hash = "sha256-TrUHW3W4fMR8esirEenKxV+cxnzjC5U0raPfutear+g=";
```

h7x4 commented on pkgs/data/icons/buuf-icon-theme/default.nix:
```diff
@@ -0,0 +1,28 @@
+{ lib, stdenvNoCC, fetchzip }:
```
I'm not able to build this package because it's not listed in `pkgs/top-level/all-packages.nix`. There is however a new paradigm on it's way in right now, where if you put the package in `pkgs/by-name/bu/buuf-icon-theme/package.nix`, it will be included in `pkgs` automatically. While it's not required just yet, feel free to try it out!

h7x4 reviewed (commented):
A few last comments

sk4rd commented:
Now you should be able to build the package 😁.

h7x4 commented:
There seems to be a bit of weirdness with the distribution of the package. Dead symlinks, pngs marked with the executable bit, xcf files for only the plus symbol, the `.git` directory included. But that's an upstream issue. Not sure if it's a blocker.

sk4rd commented:
I'm closing this pull request. I have found a better version of the buuf-icon-theme under a different name. And again, thank you for your help :-)
